api/v1alpha1/securitygroup_types.go

```diff
@@ -26,40 +26,88 @@ type RuleDirection string
 // +kubebuilder:validation:MaxLength:=16
 type Protocol string
 
+const (
+	ProtocolANY       Protocol = "any"
+	ProtocolAH        Protocol = "ah"
+	ProtocolDCCP      Protocol = "dccp"
+	ProtocolEGP       Protocol = "egp"
+	ProtocolESP       Protocol = "esp"
+	ProtocolGRE       Protocol = "gre"
+	ProtocolICMP      Protocol = "icmp"
+	ProtocolICMPV6    Protocol = "icmpv6"
+	ProtocolIGMP      Protocol = "igmp"
+	ProtocolIPIP      Protocol = "ipip"
+	ProtocolIPV6ENCAP Protocol = "ipv6-encap"
+	ProtocolIPV6FRAG  Protocol = "ipv6-frag"
+	ProtocolIPV6ICMP  Protocol = "ipv6-icmp"
```

Comment:
I'm sure there is a good reason for this, but including the ethertype in the protocol for ICMP looks problematic (e.g. what happens if I set ethertype to ipv4 and protocol to `ipv6-icmp`?).

However, my question is: are both `icmpv6` and `ipv6-icmp` really a thing?

Reply:
About icmpv6 and ipv6-icmp - They both appear at https://docs.openstack.org/api-ref/network/v2/index.html#security-group-rules-security-group-rules